src/session/session_api.c

```diff
@@ -1423,6 +1435,37 @@ __open_session(WT_CONNECTION_IMPL *conn,
 	session_ret->id = i;
 	session_ret->iface = stds;
 	session_ret->iface.connection = &conn->iface;
+	/*
+	 * Disable some methods if this is a read-only connection.  Group
+	 * them by call signature.
+	 */
+	if (F_ISSET(conn, WT_CONN_READONLY)) {
+		wt_session = &session_ret->iface;
+		wt_session->checkpoint =
+		    (int (*)(WT_SESSION *, const char *))__wt_session_notsup;
+		wt_session->compact =
+		    (int (*)(WT_SESSION *, const char *, const char *))__wt_session_notsup;
```

Comment:
@sueloverso, I don't think we're using different calling conventions (unless the compiler were to choose from multiple calling conventions, perhaps based on the function signature).  I suppose it's theoretically possible, but I've never seen a compiler that used multiple calling conventions in the same compile/build, and finding one would surprise me a lot.

And it would seem to be a roadblock to linking with 3rd party libraries?